hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/messagevalidation/V2PayloadValidator.java

```diff
@@ -124,17 +124,15 @@ protected  void validateReceivingApp(Exchange exchange, HL7Message messageObj)
 	 */
 	protected  void validateSendingFacility(Exchange exchange, HL7Message messageObj, String accessToken,
 			boolean isPharmanetMode) throws ValidationFailedException {
-		// Validate Sending facility
-		if (!StringUtil.isEmpty(messageObj.getSendingFacility())) {
-			String facilityNameFromAccessToken = getSendingFacility(accessToken);
-			if (!isPharmanetMode && StringUtil.isEmpty(messageObj.getSendingFacility())
-					|| !messageObj.getSendingFacility().equals(facilityNameFromAccessToken)) {
-				generateError(messageObj, ErrorMessage.HL7Error_Msg_FacilityIDMismatch, exchange);
-			} else if (isPharmanetMode && StringUtil.isEmpty(messageObj.getSendingFacility())
-					|| !messageObj.getSendingFacility().equals(facilityNameFromAccessToken)) {
-				generatePharmanetError(messageObj, ErrorMessage.HL7Error_Msg_FacilityIDMismatch, exchange);
+		// Validate Sending facility	
+		String facilityNameFromAccessToken = getSendingFacility(accessToken);
+		if (StringUtil.isEmpty(messageObj.getSendingFacility())) {
```

Comment:
this message is doing more than validation. It is setting the facility as well. Should setting be part of validation?

Reply:
@pankaj-cgi  This is the legacy behavior and done on HNClient side. As per the new implementation, if facility id is empty it will populate it from web token.